File: growth/assets/_tools/gen_for_cards.py
#!/usr/bin/env python3
"""Emit tailored brand-system OG cards (1200x630) for the for/<agent> pages.
Each card names the agent + its specific MCP wiring hook + the real ~60%.
Private-beta framing, no open-install copy. Writes growth/assets/_src/og-for-<slug>.svg.
Render with resvg.
"""
import html, os

OUT = os.path.join(os.path.dirname(__file__), "..", "_src")

# slug, agent name, mechanism subline, "connect" chip (mono), wiring note
CARDS = [
    ("claude-code", "Claude Code",
     "one canonical doc per query — not a CLAUDE.md reread of the repo",
     "point Claude Code at trovex over MCP"),
    ("cursor", "Cursor",
     "one canonical doc per query — not a whole-repo reread",
     "add to  .cursor/mcp.json"),
    ("windsurf", "Windsurf",
     "one canonical doc per query — not a whole-repo reread",
     "add to  mcp_config.json"),
    ("cline", "Cline",
     "one canonical doc per query — not a whole-repo reread",
     "add a remote MCP server in the MCP Servers panel"),
]

def esc(s):
    return html.escape(s, quote=False)

def card_svg(slug, agent, sub, connect):
    return f'''<svg xmlns="http://www.w3.org/2000/svg" width="1200" height="630" viewBox="0 0 1200 630" role="img" aria-label="trovex for {esc(agent)}. {esc(sub)}. About 60 percent fewer tokens per lookup. {esc(connect)}. Now in private beta.">
  <defs><style>
    .sans {{ font-family: 'Fira Sans','Segoe UI',system-ui,-apple-system,sans-serif; }}
    .mono {{ font-family: 'Fira Code','SF Mono',ui-monospace,monospace; }}
  </style></defs>
  <rect width="1200" height="630" fill="#0a0e17"/>
  <rect x="0" y="0" width="1200" height="6" fill="#22c55e"/>

  <!-- via MCP pill -->
  <g>
    <rect x="64" y="92" width="150" height="44" rx="22" fill="#0f1a14" stroke="#22c55e" stroke-width="1.5"/>
    <circle cx="90" cy="114" r="6" fill="#22c55e"/>
    <text class="mono" x="108" y="121" font-size="20" fill="#cdd6e2">via MCP</text>
  </g>

  <text class="sans" x="64" y="240" font-size="62" font-weight="700" fill="#e6edf3">trovex for <tspan fill="#22c55e">{esc(agent)}</tspan></text>
  <text class="sans" x="64" y="300" font-size="29" font-weight="400" fill="#9aa6b8">{esc(sub)}</text>

  <text class="sans" x="64" y="408" font-size="60" font-weight="700" fill="#e6edf3">~60%<tspan dx="20" font-size="38" font-weight="600" fill="#22c55e">fewer tokens per lookup</tspan></text>

  <!-- connect chip -->
  <rect x="64" y="452" width="1072" height="64" rx="12" fill="#0d1119" stroke="#1c2430" stroke-width="2"/>
  <text class="mono" x="88" y="492" font-size="24" fill="#7c8798">connect:  <tspan fill="#cdd6e2">{esc(connect)}</tspan></text>

  <line x1="64" y1="556" x2="1136" y2="556" stroke="#1c2430" stroke-width="1.5"/>
  <text class="mono" x="64" y="598" font-size="21" fill="#7c8798">trovex.dev · <tspan fill="#22c55e">now in private beta</tspan></text>
  <text class="sans" x="1136" y="598" font-size="20" fill="#5a6577" text-anchor="end">local-first · open core</text>
</svg>
'''

def main():
    for slug, agent, sub, connect in CARDS:
        path = os.path.join(OUT, f"og-for-{slug}.svg")
        with open(path, "w") as f:
            f.write(card_svg(slug, agent, sub, connect))
        print(f"wrote {path}")

if __name__ == "__main__":
    main()

File: growth/assets/_tools/gen_for_cards.mjs
// gen_for_cards.mjs — trovex.dev /for/<agent> OG cards (1200x630).
// VISUAL SYSTEM = data-editorial (memory visual-system-locked): brutalist Archivo
// display, dark #0c0d10, green = trovex's accent. POSITIONING = PUBLIC-BETA (the old
// 'private beta' framing is banned). Honest: real ~60% only; real MCP wiring per agent.
//
// Writes web/public/for/<slug>/og.png (each /for page's og:image meta).
// Deps (satori/resvg) resolve via a node_modules symlink up-tree.
// Run:  node growth/assets/_tools/gen_for_cards.mjs
import satori from "satori";
import { Resvg } from "@resvg/resvg-js";
import { readFile, writeFile, mkdir } from "node:fs/promises";
import { homedir } from "node:os";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";

const HERE = dirname(fileURLToPath(import.meta.url));
const OUT = join(HERE, "..", "..", "..", "web", "public", "for");
const REPOFONT = join(HERE, "..", "_fonts");
const F = join(homedir(), "Library", "Fonts");
const fonts = [
  { name: "Archivo", data: await readFile(join(REPOFONT, "Archivo-800.ttf")), weight: 800, style: "normal" },
  { name: "Fira Sans", data: await readFile(join(F, "FiraSans-Medium.otf")), weight: 500, style: "normal" },
  { name: "Fira Code", data: await readFile(join(F, "FiraCode-Regular.ttf")), weight: 400, style: "normal" },
];
const C = { bg: "#0c0d10", panel: "#101216", ink: "#f3f0e9", soft: "#8b8f98", subtle: "#74808f", rule: "#23262d", green: "#22c55e" };
const h = (t, p, ...k) => ({ type: t, props: { ...p, children: k.length <= 1 ? k[0] : k } });

// slug, agent, mechanism sub, MCP wiring line
const CARDS = [
  ["claude-code", "Claude Code", "One canonical doc per query, not a CLAUDE.md reread of the whole repo.", "point Claude Code at trovex over MCP"],
  ["cursor", "Cursor", "One canonical doc per query, not a whole-repo reread.", "add to .cursor/mcp.json"],
  ["windsurf", "Windsurf", "One canonical doc per query, not a whole-repo reread.", "add to mcp_config.json"],
  ["cline", "Cline", "One canonical doc per query, not a whole-repo reread.", "add a remote MCP server in the MCP Servers panel"],
];

function card(agent, sub, connect) {
  return h("div", { style: { width: "1200px", height: "630px", display: "flex", flexDirection: "column", justifyContent: "space-between", backgroundColor: C.bg, color: C.ink, padding: "58px 64px", fontFamily: "Fira Sans" } },
    h("div", { style: { display: "flex", justifyContent: "space-between", alignItems: "center", borderBottom: `1px solid ${C.rule}`, paddingBottom: "20px" } },
      h("div", { style: { display: "flex", alignItems: "center", gap: "11px", fontFamily: "Fira Code", fontSize: "26px", color: C.ink } },
        h("div", { style: { width: "16px", height: "16px", backgroundColor: C.green } }), h("span", {}, "trovex")),
      h("div", { style: { fontFamily: "Fira Code", fontSize: "19px", color: C.subtle, letterSpacing: "0.05em" } }, "via MCP")),
    h("div", { style: { display: "flex", flexDirection: "column", gap: "22px" } },
      h("div", { style: { display: "flex", flexWrap: "wrap", fontFamily: "Archivo", fontWeight: 800, fontSize: "72px", lineHeight: 1.0, letterSpacing: "-0.035em" } },
        h("span", { style: { marginRight: "0.24em" } }, "trovex for"), h("span", { style: { color: C.green } }, agent)),
      h("div", { style: { fontFamily: "Fira Sans", fontWeight: 500, fontSize: "27px", lineHeight: 1.36, color: C.soft, maxWidth: "980px" } }, sub),
      h("div", { style: { display: "flex", alignSelf: "flex-start", fontFamily: "Fira Code", fontSize: "21px", color: C.ink, backgroundColor: C.panel, border: `1px solid ${C.rule}`, borderRadius: "8px", padding: "12px 18px" } }, connect)),
    h("div", { style: { display: "flex", justifyContent: "space-between", alignItems: "center", fontFamily: "Fira Code", fontSize: "20px", borderTop: `1px solid ${C.rule}`, paddingTop: "20px" } },
      h("span", { style: { color: C.subtle } }, "~60% fewer tokens per lookup · public beta"),
      h("span", { style: { color: C.green } }, "trovex.dev")),
  );
}

for (const [slug, agent, sub, connect] of CARDS) {
  const svg = await satori(card(agent, sub, connect), { width: 1200, height: 630, fonts });
  const png = new Resvg(svg, { fitTo: { mode: "width", value: 1200 } }).render().asPng();
  const dir = join(OUT, slug);
  await mkdir(dir, { recursive: true });
  await writeFile(join(dir, "og.png"), png);
  console.log("rendered", `for/${slug}/og.png`);
}
console.log("DONE");
